src/libraries/System.IO.FileSystem/tests/Directory/Copy.cs

```diff
@@ -0,0 +1,216 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO.Tests;
+using System.Threading;
+using Xunit;
+
+namespace System.IO.FileSystem.Tests
+{
+    public sealed class Directory_CopyDirectory : FileSystemTest
+    {
+        [Fact]
+        public void Copy()
+        {
+            const string sourceDirName = $"Test_source_{nameof(Copy)}";
+            const string destDirName = $"Test_target_{nameof(Copy)}";
+            CancellationToken cancellationToken = new();
+
+            // Prerequisites
+            // Create source folder
+            if (Directory.Exists(sourceDirName))
+            {
+                Directory.Delete(sourceDirName, true);
+            }
+            Directory.CreateDirectory(sourceDirName);
```

Comment:
If you use GetTestFilePath() you don't need to clean beforehand, each path returned by GetTestFilePath() is unique. 
```suggestion
            Directory.CreateDirectory(sourceDirName);
```